<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="maximumSize">
       <size>
        <width>261</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="meshTab">
       <attribute name="title">
        <string>Mesh</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="geometryGroup">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Geometry</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="geometryTechniqueLabel">
             <property name="text">
              <string>Technique</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="geometryTechniqueMenu"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="geometryModelLabel">
             <property name="text">
              <string>Model</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="geometryModelMenu"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="meshVertexCountLabel">
             <property name="text">
              <string>Vertices</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="meshVertexCountSpin">
             <property name="minimum">
              <number>1000</number>
             </property>
             <property name="maximum">
              <number>10000000</number>
             </property>
             <property name="singleStep">
              <number>1000</number>
             </property>
             <property name="value">
              <number>100000</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QPushButton" name="generateMeshButton">
             <property name="text">
              <string>Generate Mesh</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="geometrySpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="evaluateTab">
       <attribute name="title">
        <string>Evaluate</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QGroupBox" name="shapeMeasureGroup">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Shape Measure</string>
          </property>
          <layout class="QFormLayout" name="formLayout_4">
           <item row="0" column="0">
            <widget class="QLabel" name="shapeMeasureTypeLabel">
             <property name="text">
              <string>Type</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="shapeMeasureTypeMenu"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="shapeMeasureImplLabel">
             <property name="text">
              <string>Implementation</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="shapeMeasureImplMenu"/>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QPushButton" name="evaluateMesh">
             <property name="text">
              <string>Evaluate Mesh</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="evaluateBenchmarkGroup">
          <property name="title">
           <string>Benchmarking</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QLabel" name="evaluateImplCycleCountsLabel">
             <property name="text">
              <string>Implementation Cycle Counts:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="evaluateImplCycleCountsLayout" native="true"/>
           </item>
           <item>
            <widget class="QPushButton" name="evaluateBenchmarkImplButton">
             <property name="text">
              <string>Benchmark Implementations</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="evaluateSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="smoothTab">
       <attribute name="title">
        <string>Smooth</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QGroupBox" name="smoothingGroup">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Smoothing</string>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="smoothingTechniqueLabel">
             <property name="text">
              <string>Technique</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="smoothingTechniqueMenu"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="smoothingImplementationLabel">
             <property name="text">
              <string>Implementation</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="smoothingImplementationMenu"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="smoothMinIterationLabel">
             <property name="text">
              <string>Min iteration</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="smoothMinIterationSpin">
             <property name="maximum">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="smoothMoveFactorLabel">
             <property name="text">
              <string>Move factor</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="smoothMoveFactorSpin">
             <property name="maximum">
              <double>2.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.050000000000000</double>
             </property>
             <property name="value">
              <double>0.350000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="smoothGainThresholdLabel">
             <property name="text">
              <string>Gain threshold</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="smoothGainThresholdSpin">
             <property name="decimals">
              <number>6</number>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.000001000000000</double>
             </property>
             <property name="value">
              <double>0.000100000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="2">
            <widget class="QPushButton" name="smoothMeshButton">
             <property name="text">
              <string>Smooth Mesh</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="smoothBenchmarkGroup">
          <property name="title">
           <string>Benchmarking</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPushButton" name="smoothBenchmarkImplButton">
             <property name="text">
              <string>Benchmark Implementations</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="optimisationSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>208</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="renderTab">
       <attribute name="title">
        <string>Render</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="renderingGroup">
          <property name="title">
           <string>Rendering</string>
          </property>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="renderingTypeLabel">
             <property name="text">
              <string>Type</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="renderingTypeMenu"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="shadingLabel">
             <property name="text">
              <string>Shading</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="shadingMenu"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="cameraGroup">
          <property name="title">
           <string>Camera</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="cutGroup">
          <property name="title">
           <string>Cut</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="renderingSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>318</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
